app/js/components/tags-page.js

```diff
@@ -7,10 +7,42 @@
 'use strict';
 
 import React from 'react';
-
 import TagsList from './tags-list';
 
 export default class TagsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {tags: []};
+  }
+
+  componentDidMount() {
+    this.getTagList();
+  }
+
+  componentDidUpdate() {
+    this.getTagList();
+  }
+
+  getTagList() {
+    // TODO: make this an api-client call
+    const tagsRequest = new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', '/qdb/sample/tags.json');
+
+      xhr.onload = () => {
+        if (xhr.status == 200) {
+          resolve(JSON.parse(xhr.responseText));
+        } else {
+          reject('an error has occured');
+        }
+      };
+
+      xhr.send();
+    });
+
+    tagsRequest.then(tags => this.setState({...this.state, tags}));
+  }
+
```

Comment:
Wanted to hold off until data was there...

-----Original Message-----
From: "Kristen Mills" <notifications@github.com>
Sent: ‎11/‎13/‎2015 13:20
To: "rit-sse/qdb-3.0" <qdb-3.0@noreply.github.com>
Subject: Re: [qdb-3.0] Make a request for data (#7)

In app/js/components/tags-page.js:
> +      xhr.open('GET', '/qdb/sample/tags.json');
> +
> +      xhr.onload = () => {
> +        if (xhr.status == 200) {
> +          resolve(JSON.parse(xhr.responseText));
> +        } else {
> +          reject('an error has occured');
> +        }
> +      };
> +
> +      xhr.send();
> +    });
> +
> +    tagsRequest.then(tags => this.setState({...this.state, tags}));
> +  }
> +
https://github.com/rit-sse/api-client
—
Reply to this email directly or view it on GitHub.